cachi2/core/package_managers/pip.py

```diff
@@ -1375,11 +1378,15 @@ def _assess_direct_access_requirement(cls, line: str) -> tuple[Optional[str], bo
 
         return direct_access_kind, True
 
-    @classmethod
-    def _adjust_direct_access_requirement(cls, line: str) -> tuple[str, dict[str, str]]:
+    @staticmethod
+    def _adjust_direct_access_requirement(
+        line: str, has_name_in_access_requirement: Pattern[str]
```

Comment:
`has_name_in_access_requirement` is really a bad name for an argument if you consider that now we're technically able to accept any regex pattern here (not that we're going to do that).

Reply:
Would you guys consider `has_name_in_requirement_pattern` a good option? It might be a bit long, but I think it makes it pretty clear.

Reply:
It's early for me, but maybe `pkg_name_regex`? :shrug:

I mean, I recognize that it's bad practice to put data types in variable names, but this is more about what purpose it serves than what it contains...

Reply:
It appears we're stuck on the classic "naming things is hard" problem :sweat_smile: 

As far as I can tell, this pattern doesn't have much to do with whether there is a name present or not :shrug: 
It's just a pattern to check for [direct references](https://peps.python.org/pep-0440/#direct-references).

Perhaps direct_ref_pattern?

Reply:
I believe so :man_shrugging: